### DISCUSSION TITLE
VAD Memory Leak

### DISCUSSION THREAD

**Discussion by snakers4** [snakers4/silero-vad#173] [Ideas]
# ⚠️ Important Information for VAD Python Users ⚠️

If you are using the VAD in a:

- multi-threaded or
- a multi-process application

**Do not forget to disable gradients in EACH process and / or thread.**
Otherwise memory may leak noticeably.

**Comment by snakers4** [comment]
![image](https://user-images.githubusercontent.com/12515440/155746169-251b4929-8a94-4687-a432-c6f2f5ccca61.png)

![image](https://user-images.githubusercontent.com/12515440/155746188-0ee13075-f0a8-4e4e-a73b-d55ecd6d1bbc.png)

**Reply by snakers4** [reply]
![image](https://user-images.githubusercontent.com/12515440/155945909-8db79589-ae69-4eca-86ff-be5dd8f391ce.png)

**Comment by dgoryeo** [comment]
Hi @snakers4 , is this (old) note still valid?

**Reply by snakers4** [reply]
Yes
Nothing changed for PyTorch

**Comment by Freeeeeet** [comment]
А как отключить это?
просто добавить with torch.no_grad(): при вызове модели?

**Reply by snakers4** [reply]
Do not forget to disable gradients in EACH process and / or thread.
